Remove initContainers from examples of deployment-sidecar.yaml and job-sidecar.yaml

Zeeshan50522 opened the pull request:
Why this change?

The provided example incorrectly places a sidecar container in the `initContainers` section. By definition, init containers are intended to run to completion before the main application containers start.

However, sidecar containers are meant to run alongside the main application containers for the entire duration of the pod's lifecycle. Therefore, sidecar containers should be defined in the containers section, not in the `initContainers` section.

and one more fix: Original **`Sidecar containers and Pod lifecycle`** section having inaccuracies about the usage and behavior of sidecar containers and init containers. Specifically, it incorrectly suggested that sidecar containers can be defined in the `initContainers` section and remain running for the entire lifecycle of the Pod.

Zeeshan50522 commented:
The provided example incorrectly places a sidecar container in the initContainers section. By definition, init containers are intended to run to completion before the main application containers start. 

But sidecar containers are meant to run alongside the main application containers for the entire duration of the pod's lifecycle, and sidecar containers should be defined in the containers section, not in the initContainers section.

asem-hamid commented:
I think it would be great if you could make a separate PR for the English changes. This is because it includes updates not only to the YAML file but also to the content file (sidecar-containers.md).

asem-hamid commented:
Additionally, if you remove the ` initContainers: ` part from the example but not from the content (talking about the localized version), it will be misleading. We need to make both changes at the same time

asem-hamid commented:
/retitle Remove initContainers from examples of deployment-sidecar.yaml and job-sidecar.yaml

Zeeshan50522 commented:
> Additionally, if you remove the `initContainers:` part from the example but not from the content (talking about the localized version), it will be misleading. We need to make both changes at the same time

@asem-hamid  The content has already been adjusted. I have updated the content to reflect the removal of initContainer and adjusted it in reference to the [sidecar-container](https://github.com/kubernetes/website/pull/47027/commits/2cac46d9e764c016dbf1c0d5ee0f02ef591b5ab1) reference. Please let me know if there is anything I might have missed.

dipesh-rawat commented on content/bn/examples/application/deployment-sidecar.yaml:
Please exclude the change to this non-English (Bengali) sample file.

dipesh-rawat commented on content/bn/examples/application/job/job-sidecar.yaml:
Please exclude the change to this non-English (Bengali) sample file.

dipesh-rawat commented on content/id/examples/application/deployment-sidecar.yaml:
Please exclude the change to this non-English (Indonesian) sample file.

dipesh-rawat commented on content/id/examples/application/job/job-sidecar.yaml:
Please exclude the change to this non-English (Indonesian) sample file.

dipesh-rawat commented on content/ja/examples/application/deployment-sidecar.yaml:
Please exclude the change to this non-English (Japanese) sample file.

dipesh-rawat commented on content/ja/examples/application/job/job-sidecar.yaml:
Please exclude the change to this non-English (Japanese) sample file.

dipesh-rawat commented on content/zh-cn/examples/application/deployment-sidecar.yaml:
Please exclude the change to this non-English (Chinese) sample file.

dipesh-rawat commented on content/zh-cn/examples/application/job/job-sidecar.yaml:
Please exclude the change to this non-English (Chinese) sample file.

dipesh-rawat reviewed (commented):
@Zeeshan50522 Thanks for your contribution. It appears that this PR includes changes to files across multiple languages English, Bengali, Indonesian, Japanese and Chinese. However, docs follows different processes for each localization, and we typically don't accept pull requests that impact multiple languages simultaneously. I would recommend modify only the primary website content (which means only English content) initially with this PR. After the merge of the English content, we could create separate pull request for the remaining language(s). I hope this suggestion is helpful.

dipesh-rawat commented:
Explicit hold until this PR only covers one language.
(Ok to unhold once that's done)
/hold

Zeeshan50522 commented:
Thank @dipesh-rawat for the clarification. I have reverted the changes made to the other languages and have only kept the modifications for the English content in this PR. 

Removing the hold: /unhold

asem-hamid commented:
/remove language/bn

asem-hamid commented:
/remove lang/bn

asem-hamid commented:
/remove-language bn

asem-hamid commented:
/remove-language id 
/remove-language ja
/remove-language zh

asem-hamid commented:
/unassign

asem-hamid commented:
/uncc

sftim commented:
> sidecar containers should be defined in the `containers` section, not in the `initContainers` section.

This doesn't seem right. See https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/ for our current documentation, which I believe is accurate.

/hold
- because the change looks technically incorrect.
- also we should not have removed the previous hold as the commits in this PR do also seem to touch other languages (changing a file and then reverting that bit is still a change).

sftim commented:
Make sure you've definitely tried this on a v1.30 Kubernetes cluster with the alpha support for sidecars explicitly enabled.
We're also working on a tutorial to help people try this out - see https://github.com/kubernetes/website/pull/46825 to view the early work on this.

The docs _are_ unfinished, but that doesn't mean they are wrong. I still think what the docs say is technically correct.

Zeeshan50522 commented:
yes, you are right i was running with `Kubernetes v1.28.3` with (`SidecarContainers` : false) , sorry for oversight i'm closing the PR as we have [PR](https://github.com/kubernetes/website/pull/46825) already in progress

thanks for highlighting and sorry for inconvenience
